OloEngine/src/OloEngine/Networking/Prediction/LagCompensator.cpp

```
@@ -0,0 +1,71 @@
+#include "OloEnginePCH.h"
+#include "LagCompensator.h"
+#include "OloEngine/Networking/Replication/EntitySnapshot.h"
+#include "OloEngine/Scene/Scene.h"
+#include "OloEngine/Core/Log.h"
+#include "OloEngine/Debug/Profiler.h"
+
+namespace OloEngine
+{
+    LagCompensator::LagCompensator() = default;
+
+    bool LagCompensator::PerformLagCompensatedCheck(Scene& scene, const SnapshotBuffer& history, u32 targetTick,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)